### PR TITLE
Transformed old "out of the box" reports to new reports paradigm

### PR DESCRIPTION
**Pull request by sagylell**
fixes: https://github.com/demisto/etc/issues/10319

### CODE REVIEW

**Comment by sagylell** [comment]
@Bars92 @dorsha I get "schema validation failed" error during the build.

**Comment by dorsha** [comment]
It seems like a bug in the validator script, that require stuff that are not mandatory.
@asafshen can you please check this?

image

**Comment by asafshen** [comment]
@dorsha indeed

@sagylell it seems that the schema is only support the old report schema
I'll reach to this next week, but you can try to fix this by looking at the schemas of integrations/playbooks/old reports

we use [this package](https://github.com/Grokzen/pykwalify) for schema validation
and you can run locally the validation script with `./Tests/scripts/validate_files_structure.sh`

**Comment by aviadl** [comment]
@sagylell In the mean time, you  can also disable the validation, if it is in correct (only for repots obviously)

**Comment by sagylell** [comment]
@asafshen I've disabled the reports schema test, will fix it next week, but now I'm getting another fail at './Tests/scripts/run_tests.sh'.

As far as I can see from the test's code, it only checks the playbooks and integrations, nothing related to the reports.

screen shot 2018-01-26 at 14 12 16

**Comment by dorsha** [comment]
This should fix this: https://github.com/demisto/content-test-conf/pull/39
Was a bad push to the conf file.
@sagylell try to re-run the build (there is a re-build button in circleci)

**Comment by Bars92** [comment]
Found another issue and fixed with @liorblob. Ran re-build, should be good now

**Comment by Bars92** [comment]
@sagylell are you sure on `fromVersion: "3.5"` instead of  `fromVersion: "3.5.0"` and toVersion? seems like other content items are always specifying the minor version

**Comment by sagylell** [comment]
@Bars92 I've added the minor version. Failing again because of the playbooks.

**Comment by dorsha** [comment]
@Bars92 did you finish the review?

File: Reports/report-dailyIncidentReport.json [hunk 1]
```diff
@@ -1,396 +1,821 @@
...
+    ],
+    "name": "Daily incidents",
+    "isPredefined": false,
+    "version": 0,
+    "fromDate": "0001-01-01T00:00:00Z",
+    "id": ""
+  },
+  "endingDate": "0001-01-01T00:00:00Z",
+  "sensitive": false,
+  "version": 6,
```
Comment (Bars92) [inline Reports/report-dailyIncidentReport.json]: please remove this

File: Reports/report-docx-dailyIncidentReport.json [hunk 2]
```diff
@@ -1,396 +1,821 @@
...
+    ],
+    "name": "Daily incidents",
+    "isPredefined": false,
+    "version": 0,
+    "fromDate": "0001-01-01T00:00:00Z",
+    "id": ""
+  },
+  "endingDate": "0001-01-01T00:00:00Z",
+  "sensitive": false,
+  "version": 6,
```
Comment (Bars92) [inline Reports/report-docx-dailyIncidentReport.json]: I understand you copied a created reports json. Important to note that this marks a custom made report. please remove the version from all reports

File: Reports/report-last24HoursClosedIncidentsReport.json [hunk 3]
```diff
@@ -1,406 +1,650 @@
 {
-  "id": "last24HoursClosedIncidentsReport",
+  "decoder": null,
```
Comment (Bars92) [inline Reports/report-last24HoursClosedIncidentsReport.json]: the decoder is removed here for some reason. should not be removed.

File: Reports/report-last24HoursIncidentReport.json [hunk 4]
```diff
@@ -2,431 +2,840 @@
   "id": "last24HoursIncidentReport",
   "name": "Last 24 hours incidents",
   "description": "Last 24 hours summary of incidents statistics, followed by a list of all current open incidents.",
-  "fromVersion": "2.5.0",
+  "releaseNotes": "-",
+  "fromVersion": "3.5.0",
+  "version": 10,
```
Comment (Bars92) [inline Reports/report-last24HoursIncidentReport.json]: remove version

File: Reports/report-last30DaysClosedIncidentsReport.json [hunk 5]
```diff
@@ -2,405 +2,628 @@
...
+  "recipients": [],
+  "orientation": "",
+  "paperSize": "",
+  "runOnce": false,
+  "latestReportName": "",
+  "latestReportTime": "0001-01-01T00:00:00Z",
+  "latestScheduledReportTime": "0001-01-01T00:00:00Z",
+  "nextScheduledTime": "0001-01-01T00:00:00Z",
+  "latestReportUsername": "",
+  "decoder": null,
```
Comment (Bars92) [inline Reports/report-last30DaysClosedIncidentsReport.json]: please make sure to not remove the decoder in all reports..

Comment (sagylell) [reply Reports/report-last30DaysClosedIncidentsReport.json]: This is how I got it from the "Preview" command

Comment (Bars92) [reply Reports/report-last30DaysClosedIncidentsReport.json]: it is possible you have done it before we added it. you can create a new report and check it. Just remove the changes made for decoder value

File: Reports/report-dailyIncidentReport.json (hunk 1 above)
Comment (sagylell) [reply Reports/report-dailyIncidentReport.json]: Just to make sure.. The whole 3 parameters?

**Comment by sagylell** [comment]
Done

**Comment by Bars92** [comment]
@sagylell wait with this until tomorrow, found an issue we need to address.

**Comment by Bars92** [comment]
@sagylell after https://github.com/demisto/web-client/pull/3429 you can go over all the added reports and make sure all are marked `"system": true`. This way they will be locked for editing (content update override changes), but you will be able to duplicate.

**Comment by sagylell** [comment]
Done

**Comment by Bars92** [comment]
@sagylell please notice missing some critical params. try upload some reports and check they are working as we expect. Should be good for now.

**Pull request merged**